src/modules/file-transfer-service/FileTransferService.ts

```diff
@@ -120,15 +101,22 @@ export class FileTransferService {
         }
     }
 
-    private getErrorFrom(err: any, fileId: string): Error {
-
-        if (err.isAxiosError) {
-            switch (err.response.status) {
-                case NOT_FOUND:
-                    return new Error(`File ${fileId} not found.`);
+    private handleResponseError(operation: 'upload' | 'metadata retrieval' | 'download' | 'deletion', subject: string):
+        (err: AxiosError) => never {
```

Comment:
does this mean that the following (within `{ }`) should 'never' occur?

Reply:
Never type is required by `catch` function. I started with `void` but `never` was required.